### PR TITLE
add permission for enableLink

### PR DESCRIPTION
**Pull request by xam-stephan**
please add a permission to enable the link just for some roles

### CODE REVIEW

File: configuration.php [hunk 1]
```diff
@@ -6,6 +6,7 @@
 $this->providePermission('grafana/graphconfig', $this->translate('Allow to configure graphs.'));
 $this->providePermission('grafana/debug', $this->translate('Allow to see module debug informations.'));
 $this->providePermission('grafana/showall', $this->translate('Allow access to see all graphs of a host.'));
+$this->providePermission('grafana/enableLink', $this->translate('Enable link to grafana'));
```
Comment (dnsmichi) [inline configuration.php]: I‘d suggest to keep permissions lowercase.

I would also do that for the config setting, but that‘s for a different issue.

File: library/Grafana/ProvidedHook/Grapher.php [hunk 2]
```diff
@@ -511,7 +511,7 @@ public function getPreviewHtml(MonitoredObject $object, $report = false)
             $res = $this->getMyPreviewHtml($serviceName, $hostName, $previewHtml);
 
             //do not render URLs on error or if disabled
-            if (!$res || $this->enableLink == "no") {
+            if (!$res || $this->enableLink == "no" || !$this->permission->hasPermission('grafana/enableLink')) {
```
[the remark this answers is not in the document]
Comment (Mikesch-mp) [reply library/Grafana/ProvidedHook/Grapher.php]: Yepp its correct.

**Comment by Mikesch-mp** [comment]
Please lowercase the permissions as @dnsmichi suggested and i will add it.

**Comment by xam-stephan** [comment]
ok, the permission name ist now enable-link.
To get the link the users must have the permission and the link must be enabled in the module config.
I'm not sure if you want to keep it that way

**Pull request merged**